app/models/repository/access.rb

```diff
@@ -4,6 +4,24 @@ module Repository::Access
   OPTIONS = %w[public_r public_rw private_r private_rw]
   DEFAULT_OPTION = OPTIONS[0]
 
+  def self.as_read_only(access)
+    access.split('_').first + '_r'
```

Comment:
how about using a limit to split (e.g. `...split('_', 2)`) so optimizations can kick in? I know it's probably just a short string, but why not ;)